### PR TITLE
Fixes #197: Added tests and Continuous Integration services:

### LINKED ISSUE #197: Add tests to allow integration in Travis CI and Scrutinizer
I created a fork to add some tests which can be launched by Travis CI and Scrutinizer.

I have to squash the changes in order to submit a clean PR.

### PR DESCRIPTION
**Pull request by alexislefebvre**
Added test environment:
 - AppKernel with configuration and route (see http://www.whitewashing.de/2012/02/25/symfony2_controller_testing.html for reference)
 - Dependencies in composer.json: Symfony and PHPUnit

Added DataFixtures and tests with fixtures
Added User entity (I had to use the YAML format for Doctrine, annotations didn't worked)
Web tests: added a form with validation errors assertions, added status code assertions
Added:
 - Simple command and a test for it
 - Twig view
 - Simple test file with fake authentification

Travis CI: Symfony 2.3 is in allowed failures
Scrutinizer: Composer will install Symfony 2.8.*

Added badges in README.md: Travis CI, Scrutinizer and SensioLabsInsight

You will have to update the URLs in README.md once CI services will be configured for this repository.

### CODE REVIEW

**Comment by alexislefebvre** [comment]
I'll have to fix the StyleCI issues and check that all the copyright blocks are present in each new file.

**Comment by lsmith77** [comment]
good stuff!

I just quickly fixed (via the web interface) the PHP 5.3 issue https://github.com/liip/LiipFunctionalTestBundle/commit/0981fc0a18678b50a53410496239b602dc5a355b

I have enabled travis ci for this PR.

FYI you can quickly let the styleci service fix your CS issues by clicking on "details"

**Comment by alexislefebvre** [comment]
I'll be able to improve this PR on next week.

**Comment by lsmith77** [comment]
@alexislefebvre ok great .. btw what is the value of codeclimate vs. scrutinizer/styleci?

**Comment by alexislefebvre** [comment]
@lsmith7 I prefer Scrutinizer because it runs the tests itself. AFAIK, Code
Climate requires an external service to generate and send the results (e.g.
Travis CI). This reminds me that I'll have to I'll have to remove the
unused configuration files from this PR.

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+languages:
```
Comment (lsmith77) [inline .codeclimate.yml]: should be removed

**Comment by alexislefebvre** [comment]
I have fixed the errors reported by StyleCI (I did it by hand, I didn't found how to do it automatically). All the added PHP files contain copyright text blocks.

Do I need to squash the changes in order to end up with only one commit in this PR?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    # Symfony 3.0 requires PHP >= 5.5.9
+    - php: 5.3
+      env: SYMFONY_VERSION="3.0.*"
+    - php: 5.4
+      env: SYMFONY_VERSION="3.0.*"    
+
+addons:
+  apt:
+    packages:
+      - php5-sqlite
```
Comment (lsmith77) [inline .travis.yml]: i am surprised if this is really necessary .. but I guess it also doesnt hurt

File: composer.json [hunk 3]
```diff
@@ -19,7 +19,15 @@
         "symfony/framework-bundle": "~2.3|~3.0",
         "symfony/browser-kit": "~2.3|~3.0",
         "symfony/validator": "~2.5|~3.0",
-        "doctrine/common": "~2.0"
+        "doctrine/common": "~2.0",
+        "symfony/symfony": "2.8.*"
```
Comment (lsmith77) [inline composer.json]: I guess this should be reverted?

Comment (alexislefebvre) [inline composer.json]: You're right.

File: .travis.yml (hunk 2 above)
Comment (alexislefebvre) [inline .travis.yml]: The tests are broken with Symfony 2.3 and 3.0 at this moment, so I kept the `allow_failures` part but it should be removed in the future.

Comment (lsmith77) [inline .travis.yml]: I was talking about the ``addons`` part .. the extension is afaik installed by default already

Comment (alexislefebvre) [inline .travis.yml]: Oops, OK, I didn't know. Do we keep this part?

Comment (lsmith77) [inline .travis.yml]: lets remove it

**Comment by alexislefebvre** [comment]
It should be OK now.

**Comment by lsmith77** [comment]
awesome!

**Comment by lsmith77** [comment]
FYI I fixed the travis setup and some of the issues with 2.3 and 3.0 in https://github.com/liip/LiipFunctionalTestBundle/pull/202

**Comment by alexislefebvre** [comment]
That's nice! :+1:

**Pull request merged**